### PR TITLE
fix(todoController): added request attribute check

### PR DESCRIPTION
**Pull request by ghost**
As proposed in #1

### CODE REVIEW

File: controllers/todo.controller.js [hunk 1]
```diff
@@ -15,12 +15,30 @@ module.exports = server => {
     });
 
     server.post('/todos', (req, res) => {
-        const todo = new Todo(todoId++, req.body.name, req.body.done);
+        if (!req.body.name)
+            return res.send(new errs.BadRequestError("No name for new item supplied."));
+
+        if (typeof req.body.name !== 'string')
+            return res.send(new errs.BadRequestError("name should be a string."));
+        if (req.body.done != undefined && typeof req.body.done !== 'boolean')
```
Comment (christianliebel) [inline controllers/todo.controller.js]: Could you use `!==` and `void 0` (for `undefined`), please?

File: controllers/todo.controller.js [hunk 2]
```diff
@@ -15,12 +15,30 @@ module.exports = server => {
...
+            return res.send(new errs.BadRequestError("done should be a boolean."));
+
+        const todo = new Todo(todoId++, req.body.name, req.body.done != undefined ? req.body.done : false);
```
Comment (christianliebel) [inline controllers/todo.controller.js]: `req.body.done != undefined ? req.body.done : false` can be simplified to `req.body.done || false`

File: controllers/todo.controller.js [hunk 3]
```diff
@@ -15,12 +15,30 @@ module.exports = server => {
...
         todoRepository.createOrUpdate(req.user.sub, todo);
         res.send(201, todo);
     });
 
     server.put('/todos/:id', (req, res) => {
+        if (!req.body.name)
+            return res.send(new errs.BadRequestError("No name for item supplied."));
+        if (req.body.done == undefined)
```
Comment (christianliebel) [inline controllers/todo.controller.js]: Could you use `===` and `void 0` (for `undefined`), please?

**Review by christianliebel — CHANGES_REQUESTED** [review]
Hi @eberttim! Thanks for your submission. I have three minor (i.e., non-functional) remarks regarding the code style. It would also be great if you could use _single quotes_ for string literals. Could you update your PR accordingly? The rest looks good. 👍

**Comment by ghost** [comment]
@christianliebel Thanks for your feedback!:+1: I implemented the requested changes in 0ed0f4242da37bce83c280766c0901f06580a7e4

**Review by christianliebel — APPROVED** [review]
Awesome! Thank you. 🎉

**Pull request merged**